orderer/common/multichannel/registrar_test.go

```diff
@@ -510,13 +513,13 @@ func TestBroadcastChannelSupport(t *testing.T) {
 }
 
 func TestRegistrar_JoinChannel(t *testing.T) {
```

Comment:
Add a test case that tests that the ChannelId in the channel header matches the method parameter.
Add a test case that exercises the certificate matching, and the isMemeber return value.
Add a test case that exercises bad etcdraft metadata detection.

Reply:
Originally I had just added the tests for the registrars `JoinChannel` since that's the exported function that consumers would use instead of the `validateJoinBlock` helper function. Because this PR doesnt actually process the returned values for `isMember` and `isAppChannel` in `JoinChannel` I only had the error cases tested. I can change it to just unit test the helper function though until we actually implement the rest of the `JoinChannel` behavior.